lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java

```diff
@@ -384,28 +465,38 @@ static String formatCreatedBy(String trinoVersion)
 
     private void initColumnWriters()
     {
-        ParquetProperties parquetProperties = ParquetProperties.builder()
+        ParquetProperties.Builder builder = ParquetProperties.builder()
                 .withWriterVersion(PARQUET_1_0)
-                .withPageSize(writerOption.getMaxPageSize())
-                .build();
-
+                .withMaxBloomFilterBytes(writerOption.getMaxBloomFilterSize())
+                .withPageSize(writerOption.getMaxPageSize());
+        for (ParquetWriterOptions.BloomFilterColumn column : this.bloomFilterColumns) {
+            builder.withBloomFilterEnabled(column.columnDotPath(), true);
+            if (column.NDV().isPresent()) {
+                builder.withBloomFilterNDV(column.columnDotPath(), column.NDV().orElseThrow());
+                builder.withBloomFilterFPP(column.columnDotPath(), column.FPP().orElseThrow());
+            }
+        }
         this.columnWriters = ParquetWriters.getColumnWriters(
                 messageType,
                 primitiveTypes,
-                parquetProperties,
+                builder.build(),
                 compressionCodec,
                 writerOption.getMaxPageValueCount(),
                 parquetTimeZone);
     }
 
+    record RowGroupWithBloomFilters(RowGroup rowGroup, List<Optional<BloomFilter>> bloomFilters) {}
+
     private static class FileFooter
     {
         private final MessageType messageType;
         private final String createdBy;
         private final Optional<DateTimeZone> parquetTimeZone;
 
         @Nullable
-        private ImmutableList.Builder<RowGroup> rowGroupBuilder = ImmutableList.builder();
+        private ImmutableList.Builder<RowGroupWithBloomFilters> rowGroupBuilder = ImmutableList.builder();
+        @Nullable
+        private ImmutableList<RowGroupWithBloomFilters> rowGroups;
```

Comment:
Bloom filters are not part of the file footer, they're just located adjacent to the footer. Let's not add BloomFilter to `FileFooter` class.
You can track bloom filters separately as a field in ParquetWriter (`private ImmutableList.Builder<List<Optional<BloomFilter>>> rowGroupBloomFiltersBuilder`)